stage0/src/usart.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use lpc55_pac as device;
+
+pub fn setup(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+    flexcomm: &device::flexcomm0::RegisterBlock,
+) {
+    gpio_setup(syscon, iocon);
+    flexcomm0_setup(syscon, flexcomm);
+}
+
+pub fn teardown(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+    flexcomm: &device::flexcomm0::RegisterBlock,
+) {
+    gpio_teardown(syscon, iocon);
+    flexcomm0_teardown(syscon, flexcomm);
+}
+/// Configure GPIO pin 29 & 30 for RX & TX respectively, as well as
+/// digital mode.
+fn gpio_setup(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+) {
+    // IOCON: enable clock & reset
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().enable());
+    syscon.presetctrl0.modify(|_, w| w.iocon_rst().released());
+
+    // GPIO: enable clock & reset
+    syscon.ahbclkctrl0.modify(|_, w| w.gpio0().enable());
+    syscon.presetctrl0.modify(|_, w| w.gpio0_rst().released());
+
+    // configure GPIO pin 29 & 30 for RX & TX respectively, as well as
+    // digital mode
+    iocon
+        .pio0_29
+        .write(|w| w.func().alt1().digimode().digital());
+    iocon
+        .pio0_30
+        .write(|w| w.func().alt1().digimode().digital());
+
+    // disable IOCON clock
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().disable());
```

Comment:
probably not worth turning off IOCON since the ROM ends up leaving it on for us anyway

Reply:
fixed